### PR TITLE
Add some questions and answers to the FAQ

### PR DESCRIPTION
**Pull request by gridbugs**
This documents some cases I've encountered that I found confusing.

### CODE REVIEW

**Review by christinerose — APPROVED** [review]
Looks great! I made some basic formatting/grammar edits to avoid filling up the conversation with a comma here and there. 🙂

**Comment by gridbugs** [comment]
> Looks great! I made some basic formatting/grammar edits to avoid filling up the conversation with a comma here and there. slightly_smiling_face

Excellent, thanks! I think I tend to overuse commas!

File: doc/faq.rst [hunk 1]
```diff
@@ -122,3 +136,125 @@ file:
...
+  (library
+   (public_name my_library)
+   ...)
+
+You can build this library on its own by running the following from the project
+root directory:
+
+.. code:: bash
+
+   $ dune build %{cmx:src/foo/my_library}
```
Comment (rgrinberg) [inline doc/faq.rst:161]: Better is `%{cmxa:..}` or `%{cma:..}` to build the entire archive rather than the object for a specific module.

Comment (gridbugs) [reply doc/faq.rst]: changed to `cmxa`

File: doc/faq.rst [hunk 2]
```diff
@@ -122,3 +136,125 @@ file:
...
+
+   $ dune build %{cmxa:src/foo/my_library}
+
+Note that the path (``src/foo`` in the example above) is relative to the current
+directory - not the project root. If the library defines a ``name`` distinct from
+its ``public_name`` then that can be used interchangeably with the ``public_name``
+in this command.
+
+Files and Directories Whose Names Begin with "." (Period) are Ignored by ``source_tree``
+======================================================================================
```
Comment (rgrinberg) [inline doc/faq.rst:169]: Title underline is too short (see the docs CI)

Comment (gridbugs) [reply doc/faq.rst]: oops, updated the title without updating the underline. fixed

**Comment by christinerose** [comment]
> Excellent, thanks! I think I tend to overuse commas!

No worries at all! It's very common to overuse commas, as people tend to put them where they'd take a breath or a short pause. You don't have to worry about nitpicky things like that. It's why I'm here, and I inexplicably enjoy it. 😄

**Pull request merged**